WGLMakie/src/particles.jl

```diff
@@ -165,6 +167,24 @@ function serialize_three(fta::NoDataTextureAtlas)
     return tex
 end
 
+function add_f32c_scale!(uniforms, scene, plot, f32c)
```

Comment:
this looks like the same function as in GLMakie? Can we share them, and also put the huge comment from GLMakie there?